paimon-core/src/main/java/org/apache/paimon/utils/BranchManager.java

```diff
@@ -80,6 +82,62 @@ public Path branchPath(String branchName) {
         return new Path(getBranchPath(tablePath, branchName));
     }
 
+    /** Create empty branch. */
+    public void createBranch(String branchName) {
+        checkArgument(
+                !branchName.equals(DEFAULT_MAIN_BRANCH),
+                String.format(
+                        "Branch name '%s' is the default branch and cannot be used.",
+                        DEFAULT_MAIN_BRANCH));
+        checkArgument(!StringUtils.isBlank(branchName), "Branch name '%s' is blank.", branchName);
+        checkArgument(!branchExists(branchName), "Branch name '%s' already exists.", branchName);
+        checkArgument(
+                !branchName.chars().allMatch(Character::isDigit),
+                "Branch name cannot be pure numeric string but is '%s'.",
+                branchName);
+        try {
+            fileIO.mkdirs(branchPath(branchName));
```

Comment:
The schema should be copied into the branch dir, just like creating a table without any data. Thus we can write to this table(branch) directly. Otherwise I think this new branch couldn't be write into normally after being created.

Reply:
> I think the latest schema could be copied and we could add UTs for writing to an empty branch after creating it.

Complete unit testing after merging # 2950